docs/network.md

```diff
@@ -0,0 +1,65 @@
+# 1. Overview
+Near Protocol uses its own implementation of a custom peer-to-peer network
+Peers who join network are represented by nodes and connected between them by edges.
+
+The purpose of this document is to describe inner workings of `near-network` package;
+and to be used as reference by future engineers to understand network code without any prior knowledge.
+
+# 2. Code structure
+`near-network` runs on top of `actor` framework called `Actix` (https://actix.rs/docs/).
+Code structure is split between 4 actors `PeerManagerActor`, `PeerActor`, `RoutingTableActor`, `EdgeValidatorActor`
+
+### 2.1 `EdgeValidatorActor` (currently called `EdgeVerifierActor` in the code (TODO rename))
+`EdgeValidatorActor` runs on separate thread.
+The purpose of this `actor` is to validate `edges`, where each `edge` represents a connection between two peers,
+and it's signed with a cryptographic signature of both parties.
+The process of edge validation involves verifying cryptographic signatures, which can be quite expensive,
+and therefore was moved to another thread.
+
+Responsibilities:
+- validating edges by checking whenever cryptographic signatures match.
+
+### 2.2 `RoutingTableActor`
+`RoutingTableActor` maintain view of the `P2P network` represented by set of nodes and edges.
+
+In case a message needs to be sent between two nodes, let's call them `A` and `B`.
+If nodes `A` and `B` are connected then the message san be sent directly.
+However, if there aren't a node `C` needs to be chosen to forward message from `A` to `B`.
```

Comment:
I'd say that: if there is no direct connection, 'RoutingTableActor' is responsible for picking the best path between the two.